src/components/Footer.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+import { Status } from '../types/Status';
+import { Todo } from '../types/Todo';
+import { deleteTodo } from '../api/todos';
+import { useTodosContext } from './useTodosContext';
+
+export const Footer: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    query,
+    setQuery,
+    setLoadingTodosIds,
+    handleError,
+    setIsInputFocused,
+  } = useTodosContext();
+  const completedIds = useMemo(() => {
+    return todos.filter(item => item.completed === true).map(item => item.id);
```

Comment:
```suggestion
    return todos
      .filter(item => item.completed)
      .map(item => item.id);
```
Can be simplified. Btw, you can use reduce method instead of two methods